includes/REST_API/Stories_Lock_Controller.php

```diff
@@ -0,0 +1,366 @@
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has read access for the item, WP_Error object otherwise.
+	 */
+	public function get_item_permissions_check( $request ) {
+		return $this->parent_controller->update_item_permissions_check( $request );
+	}
+
+	/**
+	 * Checks if a given request has access to update a lock.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to update the item, WP_Error object otherwise.
+	 */
+	public function update_item_permissions_check( $request ) {
+		return $this->parent_controller->update_item_permissions_check( $request );
+	}
+
+	/**
+	 * Checks if a given request has access to delete a lock.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return true|WP_Error True if the request has access to delete the item, WP_Error object otherwise.
+	 */
+	public function delete_item_permissions_check( $request ) {
+		$result = $this->parent_controller->update_item_permissions_check( $request );
+		if ( is_wp_error( $result ) ) {
+			return $result;
+		}
+
+		$lock = $this->get_lock( $request['id'] );
+		if ( is_array( $lock ) && isset( $lock['user'] ) && get_current_user_id() !== (int) $lock['user'] ) {
+			return new WP_Error(
+				'rest_cannot_delete_others_lock',
+				__( 'Sorry, you are not allowed delete others lock.', 'web-stories' ),
+				[ 'status' => rest_authorization_required_code() ]
+			);
+		}
+
+		return true;
+	}
+
+	/**
+	 * Prepares a single lock output for response.
+	 *
+	 * @param array|false     $lock Lock value, default to false is not set.
+	 * @param WP_REST_Request $request Request object.
+	 *
+	 * @return WP_REST_Response|WP_Error Response object.
+	 */
+	public function prepare_item_for_response( $lock, $request ) {
+		$nonce = wp_create_nonce( 'wp_rest' );
+		$data  = [
+			'locked' => false,
+			'nonce'  => $nonce,
+		];
+
+		if ( $lock ) {
+			/** This filter is documented in wp-admin/includes/ajax-actions.php */
+			$time_window = apply_filters( 'wp_check_post_lock_window', 150 );
+
+			if ( $lock['time'] && $lock['time'] > time() - $time_window ) {
+				$data = [
+					'locked' => true,
+					'time'   => $lock['time'],
+					'user'   => (int) $lock['user'],
+					'nonce'  => $nonce,
+				];
+			}
+		}
+		// Wrap the data in a response object.
+		$context = ! empty( $request['context'] ) ? $request['context'] : 'view';
+		$data    = $this->add_additional_fields_to_object( $data, $request );
+		$data    = $this->filter_response_by_context( $data, $context );
+
+		$response = rest_ensure_response( $data );
+		if ( ! is_wp_error( $response ) ) {
+			$response->add_links( $this->prepare_links( $lock, $request['id'] ) );
+		}
+
+		/**
+		 * Filters the lock data for a response.
+		 *
+		 * The dynamic portion of the hook name, `$this->post_type`, refers to the post type slug.
+		 *
+		 * @since 4.7.0
```

Comment:
```suggestion
		 * @since 1.6.0
```

Copied and pasted from core.